ios/RNSentry.m

```diff
@@ -119,29 +121,33 @@ + (BOOL)requiresMainQueueSetup {
               });
 }
 
-RCT_EXPORT_METHOD(sendEvent:(NSDictionary * _Nonnull)event
+RCT_EXPORT_METHOD(captureEnvelope:(NSDictionary * _Nonnull)envelopeDict
                   resolve:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject)
 {
-    if ([NSJSONSerialization isValidJSONObject:event]) {
-        NSData *jsonData = [NSJSONSerialization dataWithJSONObject:event
-                                                           options:0
-                                                             error:nil];
-        
-        SentryEvent *sentryEvent = [[SentryEvent alloc] initWithJSON:jsonData];
-#if DEBUG
-        [SentrySDK captureEvent:sentryEvent];
-#else
-        // We check this against the dictionary
-        if ([event[@"level"] isEqualToString:@"fatal"]) {
-            // captureEvent queues the event for submission, so storing to disk happens asynchronously
-            // We need to make sure the event is written to disk before resolving the promise.
-            // This could be replaced by SentrySDK.flush() when available.
-            [[[[SentrySDK currentHub] getClient] fileManager] storeEvent:sentryEvent];
-        } else {
-            [SentrySDK captureEvent:sentryEvent];
-        }
-#endif
+    if ([NSJSONSerialization isValidJSONObject:envelopeDict]) {
+        SentrySdkInfo *sdkInfo = [[SentrySdkInfo alloc] initWithDict: envelopeDict[@"header"]];
+        SentryId *eventId = [[SentryId alloc] initWithUUIDString:envelopeDict[@"header"][@"event_id"]];
+        SentryEnvelopeHeader *envelopeHeader = [[SentryEnvelopeHeader alloc] initWithId:eventId andSdkInfo:sdkInfo];
+
+        NSData *envelopeItemData = [NSJSONSerialization dataWithJSONObject:envelopeDict[@"payload"] options:0 error:nil];
```

Comment:
`m`: I know this was already there, but maybe it is good to not ignore the error here. I guess if this happens there's not much we can do and the envelope anyway can't be sent. So we could do something like this:
```
NSError *error;
[NSJSONSerialization dataWithJSONObject:envelopeDict[@"payload"] options:0 error:&error];

if (nil != error) {
  reject(@"SentryReactNative", @"Cannot serialize event", nil);
}
```